modules/programs/pubs.nix

```
@@ -0,0 +1,58 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.pubs;
+
+  configFile = pkgs.writeText ".pubsrc" "${cfg.config}";
+
+in {
+  options.programs.pubs = {
+    enable = mkEnableOption "pubs";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.pubs;
+      defaultText = literalExpression "pkgs.pubs";
+      description = "The package to use for the pubs script.";
+    };
+
+    config = mkOption {
+      type = types.lines;
+      default = "";
+      example = ''
+        [main]
+        pubsdir = ${config.home.homeDirectory}/.pubs
+        docsdir = ${config.home.homeDirectory}/.pubs/doc
```

The example must not have a dependency on your configuration since it can cause unnecessary documentation builds. I think you want
```suggestion
        pubsdir = ''${config.home.homeDirectory}/.pubs
        docsdir = ''${config.home.homeDirectory}/.pubs/doc
```
instead.